### PR TITLE
438 ci npm doc

### PR DESCRIPTION
**Pull request by 4ydan**
## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by 4ydan** [comment]
What do we think about the idea of a root level Makefile with the ability to create certain docs, maybe also later for the REST api doc and other stuff? This way we could remove some of the scripts inside the ci folder, which could be also interesting for non ci purposes. Could also add pre-commit setup to the makefile.

**Comment by markus2330** [comment]
In general I don't like to have too much on top-level but one Makefile to build everything sounds like a good idea.

**Comment by 4ydan** [comment]
Can I add myself to the Authors in this PR or should I create a new one?

**Comment by 4ydan** [comment]
@markus2330 Sorry, I executed the wrong command in the pipeline, can we terminate it please.

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
Sorry, closed by accident.

**Comment by 4ydan** [comment]
> Also do I add the built storybook to deployment PR/dev and production?

Ok

> I hope you already got access so that you could terminate it yourself. If something like this happens to someone else, please terminate/restart it for them.

Maybe I missed it, but I dont have access yet.

**Comment by 4ydan** [comment]
Ok the static storybook is now build and unstashed in PR, Master and Release.

File: ci/build-scripts/build-storybook.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+echo "checking storybook"
```
Comment (markus2330) [inline ci/build-scripts/build-storybook.sh:3]:
```suggestion
echo "building storybook"
```

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 
 ## 0.2.1 - UNRELEASED
 
-- _()_
+- CI: Story-book build and stashed in Jenkins pipeline PR/Master/Release #438 _(4ydan)_
```
Comment (markus2330) [inline doc/CHANGELOG.md:11]: please use a random line (see text above), then the rebase which is needed now could've been avoided.

**Review by markus2330 — APPROVED** [review]
LGTM, great to have this!

**Comment by markus2330** [comment]
@lukashartl pls have a look. Does it include everything you would need for deployment?

**Comment by 4ydan** [comment]
I mixed up npm doc with the storybook, now everything should be here.

**Comment by markus2330** [comment]
Is this a disc-full error or is typeshare missing here?

```
+ cd frontend
+ npm run generate-api-types

> permaplant@0.2.1 generate-api-types
> cd ../backend && typeshare ./ --lang=typescript --output-file=../frontend/src/bindings/definitions.ts

sh: 1: typeshare: not found
script returned exit code 127
```

PR also needs rebase.

**Comment by 4ydan** [comment]
> Is this a disc-full error or is typeshare missing here?

Yes, its typeshare missing.

**Review by fel115 — COMMENTED** [review]
Everything looks great, and I have no complaints or suggestions. Well done!

**Pull request closed without merging**